### PR TITLE
support metrics-only mode in ResponsiveKafkaStreams

### PR DESCRIPTION
**Pull request by agavra**
the new config (`responsive.compatibility.mode = METRICS_ONLY`) effectively disables the following:
- config validation
- responsive restore consumer
- responsive global consumer
- connecting to responsive storage

### CODE REVIEW

File: kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java [hunk 1]
```diff
@@ -37,6 +37,11 @@ public class ResponsiveConfig extends AbstractConfig {
 
   // ------------------ connection configurations -----------------------------
 
+  public static final String COMPATIBILITY_MODE_CONFIG = "responsive.compatibility.mode";
+  private static final String COMPATIBILITY_MODE_DOC = "This configuration enables running Responsive "
+      + "in compatibility mode, disabling certain features.";
```
Comment (rodesai) [inline kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java:42]: compatibility is a strange word for this. Maybe "instrumentation"?

File: kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/EndOffsetsPoller.java [hunk 2]
```diff
@@ -63,8 +64,9 @@ public class EndOffsetsPoller {
 
   public EndOffsetsPoller(
       final Map<String, ?> configs,
-      final ResponsiveMetrics metrics
-  ) {
+      final ResponsiveMetrics metrics,
+      final KafkaClientSupplier clientSupplier
```
Comment (rodesai) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/EndOffsetsPoller.java:68]: nit: can you just have this accept an admin client? I feel like KafkaClientSupplier should only be called by streams.

**Review by rodesai — APPROVED** [review]
Mostly LGTM. Couple minor suggestions inline.

File: kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java (hunk 1 above)
Comment (agavra) [reply kafka-client/src/main/java/dev/responsive/kafka/api/config/ResponsiveConfig.java:42]: the idea was whether it should be compatible with existing stream deployments or not - basically I wanted to communicate that if you run it in "compatibility mode" you will not be able to use responsive stores.

**Pull request merged**